code/game/mecha/combat/combat.dm

```diff
@@ -4,6 +4,7 @@
 	damage_absorption = list("brute"=0.7,"fire"=1,"bullet"=0.7,"laser"=0.85,"energy"=1,"bomb"=0.8)
 
 /obj/mecha/combat/CheckParts(atom/holder)

```

Comment:
Checkparts is supposed to be univerlas proc for everything to check its contents, get useful shit form there and incorporate inside for some purpose
Its not hardcoded to be crafting-only, mech building and machine building can use it too